kube-core/src/object.rs

```diff
@@ -132,6 +136,42 @@ impl<'a, T: Clone> IntoIterator for &'a mut ObjectList<T> {
     }
 }
 
+#[derive(Deserialize)]
+struct DynamicList {
+    #[serde(flatten, deserialize_with = "deserialize_v1_list_as_default")]
+    types: TypeMeta,
+
+    #[serde(default)]
+    metadata: ListMeta,
+
+    #[serde(
+        deserialize_with = "deserialize_null_as_default",
+        bound(deserialize = "Vec<DynamicObject>: Deserialize<'de>")
+    )]
+    items: Vec<DynamicObject>,
+}
+
+impl<'de, T: DeserializeOwned + Clone> serde::Deserialize<'de> for ObjectList<T> {
+    fn deserialize<D: Deserializer<'de>>(d: D) -> Result<ObjectList<T>, D::Error> {
+        let DynamicList {
+            types,
+            metadata,
+            mut items,
+        } = DynamicList::deserialize(d)?;
+        let mut resources = vec![];
+        for o in items.iter_mut() {
+            o.types = Some(types.clone().singular());
```

Comment:
Sorry for the ping @clux, just wanted to ask about this change. `DynamicObject` lists are affected by https://github.com/kubernetes/kubernetes/issues/80609, and this performs correction.

`k8s-openapi` already does this for all generated objects during individual object deserialization, so the issue appears only with unstructured resource list.

Is this something which can be done in this instance? Issue may not go away for a while.

Reply:
If it is passed directly, then largely it is available, but it is on dev to store it and pass it around. Internals don't do that, and returned objects in the list as `Vec<DynamicObject>` always have empty type meta as the ListMeta goes out of scope, and the only clue is stored there.

The story is different in static types, as it is custom serialization which fixes that: https://github.com/Arnavion/k8s-openapi/blob/30a268f669650450fd4391d3785a078dbbf1d1ee/src/v1_32/api/core/v1/pod.rs#L153-L154

Reply:
This is now removed, for context: https://github.com/kube-rs/kube/pull/1692#discussion_r1998653497